packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -683,27 +693,37 @@ function ResizableTableColumnHeader(props) {
         layout.onColumnResizeStart(column.key);
         setIsInResizeMode(true);
         break;
+      default:
+        if (onAction) {
+          onAction(key, column.key);
```

Comment:
I think it might be confusing to have two different keys here. Doesn't match our other onAction APIs (e.g. Menu). I guess you added this because it might be common to reuse the same menu across multiple columns. We could make this an object with two properties so it's clear which is which, but we haven't done that before either. Or we could keep it one key, and let the user bind their event handler to the column key themselves if needed as they would with other APIs. Thoughts?

Reply:
Yeah, it was easier with the shared menu's and menu handler. Maybe I could just name them better in the TS and descriptions?
Otherwise an object seems like the better option in terms of tracking names.

Letting users bind their event handler to the column key themselves would be the most consistent though. Just seems like extra setup and maybe they want to memo them, making it even more setup.

Reply:
I like the Table-specific menu that wraps Menu idea that @LFDanLu mentioned.

Reply:
I prefer using Menu because we make a big effort to use only one version/implementation of any given component. It's why we use slots to pass props to components like Divider and Heading inside a Dialog instead of having DialogDivider and DialogHeading.

Maybe it makes more sense to make a little utility to help people create the Menus? I didn't find it to be that much work to create them though.

Reply:
In that case I think using Menu is fine. I don't think a utility will be necessary so long as we have an example in the docs since as you said it shouldn't be too much work for the user to create it